app/src/main/java/com/example/recipeapp/fragments/BarcodeScanFragment.java

```diff
@@ -0,0 +1,268 @@
+            @Override
+            public void onClick(View v) {
+                if (imageBitmap != null) {
+                    progressDialog.setMessage("Scanning Barcode...");
+                    progressDialog.show();
+                    Log.i(TAG, "Image bitmap: " + imageBitmap);
+                    InputImage image = InputImage.fromBitmap(imageBitmap, 0);
+                    scanBarcodes(image);
+                }
+            }
+        });
+    }
+
+    private void scanBarcodes(InputImage image) {
+        Log.i(TAG, "Scanning Barcode...");
+        BarcodeScannerOptions options =
+                new BarcodeScannerOptions.Builder()
+                        .setBarcodeFormats(
+                                Barcode.FORMAT_QR_CODE,
+                                Barcode.FORMAT_AZTEC,
+                                Barcode.FORMAT_ALL_FORMATS,
+                                Barcode.FORMAT_UPC_A)
+                        .build();
+        BarcodeScanner scanner = BarcodeScanning.getClient();
+        Task<List<Barcode>> result = scanner.process(image)
+                .addOnSuccessListener(new OnSuccessListener<List<Barcode>>() {
+                    @Override
+                    public void onSuccess(List<Barcode> barcodes) {
+                        Log.i(TAG, "Success barcode, ");
+                        for (Barcode barcode : barcodes) {
+                            Log.i(TAG, barcode.toString());
+                            Rect bounds = barcode.getBoundingBox();
+                            Point[] corners = barcode.getCornerPoints();
+
+                            int valueType = barcode.getValueType();
+                            // See API reference for complete list of supported types
+                            switch (valueType) {
+                                case Barcode.TYPE_WIFI:
+                                    String ssid = barcode.getWifi().getSsid();
+                                    String password = barcode.getWifi().getPassword();
+                                    int type = barcode.getWifi().getEncryptionType();
+                                    Log.i(TAG, "Barcode: " + ssid + " " + password);
+                                    break;
+                                case Barcode.TYPE_URL:
+                                    String title = barcode.getUrl().getTitle();
+                                    String url = barcode.getUrl().getUrl();
+                                    Log.i(TAG, "Barcode: " + title + " " + url);
+                                    break;
+                                case Barcode.TYPE_PRODUCT:
+                                    String rawValue = barcode.getRawValue();
+                                    Log.i(TAG, "Barcode: " + rawValue);
+                                    binding.tvRawvalue.setText("Raw Value: " + rawValue);
+                                    break;
+
+                            }
```

Comment:
Might want to route this logic to a method that accepts a list of barcodes as an argument.